multimodal/src/autogluon/multimodal/learners/semantic_segmentation.py

```diff
@@ -272,11 +426,104 @@ def save_segmentation_result(self, pred: Iterable, data: Union[pd.DataFrame, Dic
         txt_path = os.path.join(result_path, "result.txt")
         os.makedirs(mask_path, exist_ok=True)
         for image_pred, image_name in zip(pred, image_names):
-            mask = Image.fromarray(np.squeeze(image_pred, axis=0))
-            per_mask_path = os.path.join(mask_path, os.path.basename(image_name))
-            mask.save(per_mask_path)
+            if self._output_shape == 1:
+                mask = Image.fromarray(np.squeeze(image_pred, axis=0))
+                per_mask_path = os.path.join(mask_path, os.path.basename(image_name))
+                mask.save(per_mask_path)
+            else:
+                masks = []
+                classes = np.unique(image_pred)
+                for class_id in classes:
+                    if class_id == 0:  # bg
+                        continue
+                    masks.append(image_pred == class_id)
+
+                for mask in masks:
+                    show_mask(mask, plt.gca())
+                # mask = Image.fromarray(image_pred, mode="P")  # multi-class
+                mask_name = ""
+                for i in os.path.basename(image_name).split(".")[:-1]:
+                    mask_name += i
+                per_mask_path = os.path.join(mask_path, os.path.basename(image_name))
+                plt.axis("off")
+                plt.savefig(per_mask_path, bbox_inches="tight", dpi=300, pad_inches=0.0)
+
             results.append([image_name, per_mask_path])
 
         result_df = pd.DataFrame(results, columns=["image", "mask"])
         result_df.to_csv(txt_path, index=False)
         return result_df
+
+    @classmethod
+    def load(
```

Comment:
Most code is duplicate. Maybe we can just add the output_shape in `base.py`'s `get_loss_func_per_run` in `load`.